.github/workflows/release.yml

```
@@ -206,20 +206,32 @@ jobs:
         with:
           shared-key: release
 
+      - name: Clear cargo registry cache
+        run: |
+          rm -rf ~/.cargo/registry/index/*
+          rm -rf ~/.cargo/registry/cache/*
+          cargo update --workspace
+
```

I think, maybe removing the previous step might help it? As we set the cache just to remove it next step. Please try experimenting with it.

```suggestion
      - name: Update dependencies
        run: cargo update --workspace
```